File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        lp = 0
        rp = len(height) - 1

        mx_water = 0
        while lp < rp:
            w = rp - lp
            ht = min(height[lp], height[rp])
            c_water = w * ht
            mx_water = max(mx_water, c_water)

            if height[lp] < height[rp]:
                lp+=1
            else:
                rp-=1
        
        return mx_water
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int mx_water = 0;
        int lp = 0, rp = n - 1;
        while(lp < rp)
        {
            int w = rp - lp;
            int ht = min(height[lp], height[rp]);
            int c_water = w * ht;
            mx_water = max(mx_water, c_water);

            height[lp] < height[rp] ? lp++ : rp--;
        }   
        return mx_water;
    }
};